### PR TITLE
Pull Request for Issue 1964: Create a SGMContinuousScanController

### PR DESCRIPTION
**Pull request by iainworkman**
- Moved the SGM specific logic from AMContinuousScanActionController to SGMContinuousScanController, to remove the CLS dependency from an AM class
- Made SGMContinuousScanController the parent of SGMXASScanController, SGMLineScanController and SGMMapScanController.

**NOTE:** I am merging this into SGMUpgrade_mergedMaster as it was only by branching from there that I could test whether it built the suite.

### CODE REVIEW

**Comment by iainworkman** [comment]
Hey @XuetaoLiu-CLS , would you mind reviewing this?
Thanks.

File: source/acquaman/AMContinuousScanActionController.cpp [hunk 1]
```diff
@@ -277,508 +273,11 @@ void AMContinuousScanActionController::fillDataMaps(AMAgnosticDataAPIDataAvailab
 	Q_UNUSED(message)
 }
 
-bool AMContinuousScanActionController::generateAnalysisMetaInfo()
```
Comment (XuetaoLiu-CLS) [inline source/acquaman/AMContinuousScanActionController.cpp]: The following functions are moved to `SGMContinuousScanController`

**Comment by XuetaoLiu-CLS** [comment]
Looks good. It's good to move the SGM specific codes out of the AM classes.

Time Estimate: 10 minutes

File: source/acquaman/AMContinuousScanActionController.cpp (hunk 1 above)
Comment (iainworkman) [inline source/acquaman/AMContinuousScanActionController.cpp]: In cases where I thought the function might have some applicability in a
general sense I left them in the AM version. I moved the ones that made
specific use of the scaler or Amptek.

**Comment by dretrex** [comment]
Looks good.
Time Estimate: 3 minutes

**Pull request merged**